device/thunder_ripple_sdk/src/processors/thunder_package_manager.rs

```diff
@@ -640,12 +671,14 @@ impl ThunderPackageManagerRequestProcessor {
 
         match thunder_resp {
             Ok(handle) => {
-                let operation = Operation::new(
-                    AppsOperationType::Install,
-                    app.id,
-                    AppData::new(app.version),
-                );
+                let operation_timer = operation.timer.clone().stop();
                 Self::add_or_remove_operation(state.clone(), handle.clone(), operation);
+                stop_and_send_service_metrics_timer(
```

Comment:
I don't think you want to stop the operation timer here, you want to stop it when we get a notification that an operation has completed (successfully or not). That happens [here](https://github.com/rdkcentral/Ripple/blob/main/device/thunder_ripple_sdk/src/processors/thunder_package_manager.rs#L387), but the other situation you have to account for is that sometimes an operation will complete before we even get the `Ok(handle)` response from thunder in this block. Because of this I wonder if instead you want to conditionally start/stop the operation timer in [add_or_remove_operation()](https://github.com/rdkcentral/Ripple/blob/main/device/thunder_ripple_sdk/src/processors/thunder_package_manager.rs#L464), which figures out whether to start the operation time-out timer depending on whether the operation is being added or removed (see the comment tombstone).